x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elser/ElserMlNodeService.java

```diff
@@ -0,0 +1,167 @@
+import org.elasticsearch.xpack.inference.services.MapParsingUtils;
+
+import java.util.List;
+import java.util.Map;
+
+import static org.elasticsearch.xpack.core.ml.inference.assignment.AllocationStatus.State.STARTED;
+import static org.elasticsearch.xpack.inference.services.MapParsingUtils.removeFromMapOrThrowIfNull;
+
+public class ElserMlNodeService implements InferenceService {
+
+    public static final String NAME = "elser_mlnode";
+
+    private static final String ELSER_V1_MODEL = ".elser_model_1";
+
+    public static ElserMlNodeModel parseConfig(
+        boolean throwOnUnknownFields,
+        String modelId,
+        TaskType taskType,
+        Map<String, Object> settings
+    ) {
+        Map<String, Object> serviceSettingsMap = removeFromMapOrThrowIfNull(settings, Model.SERVICE_SETTINGS);
+        Map<String, Object> taskSettingsMap = removeFromMapOrThrowIfNull(settings, Model.TASK_SETTINGS);
+
+        var serviceSettings = serviceSettingsFromMap(serviceSettingsMap);
+        var taskSettings = taskSettingsFromMap(taskType, taskSettingsMap);
+
+        if (throwOnUnknownFields == false) {
+            throwIfNotEmptyMap(settings);
+            throwIfNotEmptyMap(serviceSettingsMap);
+            throwIfNotEmptyMap(taskSettingsMap);
+        }
+
+        return new ElserMlNodeModel(modelId, taskType, NAME, serviceSettings, taskSettings);
+    }
+
+    private final OriginSettingClient client;
+
+    public ElserMlNodeService(Client client) {
+        this.client = new OriginSettingClient(client, ClientHelper.INFERENCE_ORIGIN);
+    }
+
+    @Override
+    public Model parseConfigStrict(String modelId, TaskType taskType, Map<String, Object> config) {
+        return parseConfig(true, modelId, taskType, config);
+    }
+
+    @Override
+    public ElserMlNodeModel parseConfigLenient(String modelId, TaskType taskType, Map<String, Object> config) {
+        return parseConfig(false, modelId, taskType, config);
+    }
+
+    @Override
+    public void start(Model model, ActionListener<Boolean> listener) {
+        if (model instanceof ElserMlNodeModel == false) {
+            listener.onFailure(new IllegalStateException("not an elser model")); // TODO
+            return;
+        }
+
+        if (model.getTaskType() != TaskType.SPARSE_EMBEDDING) {
+            listener.onFailure(new IllegalStateException("not an elser model")); // TODO
+            return;
+        }
+
+        var elserModel = (ElserMlNodeModel) model;
+        var serviceSettings = elserModel.getServiceSettings();
+
+        var startRequest = new StartTrainedModelDeploymentAction.Request(ELSER_V1_MODEL, model.getModelId());
+        startRequest.setNumberOfAllocations(serviceSettings.getNumAllocations());
+        startRequest.setThreadsPerAllocation(serviceSettings.getNumThreads());
+        startRequest.setWaitForState(STARTED);
+
+        client.execute(
+            StartTrainedModelDeploymentAction.INSTANCE,
```

Comment:
If the ML plugin is disabled, or if the NLP feature within it is disabled, this will give an unfriendly error message. It will be something along the lines of "[cluster:admin/xpack/ml/trained_models/deployment/start] action does not exist". We should map that to something friendlier by specifically looking for `ActionNotFoundTransportException` (please confirm that's the right one) and rethrowing as something like, "your cluster is not set up for inference on ML nodes - please contact your administrator".

Reply:
I'm tracking this issue, one thing that is apparent is that we will require tests to show the API works when there are no ml nodes in the cluster. I will follow up with those tests and address this at the same time.

Reply:
There are two separate scenarios:

1. When the APIs exist but there are no ML nodes in the cluster
2. When the APIs do not exist, because either the ML plugin is disabled or the NLP feature within it is disabled

For the case of no ML nodes (or no capacity left on existing ML nodes) there are then three further scenarios:

1. Autoscaling enabled and no limits preventing scale up
2. Autoscaling enabled but the autoscaling limit is reached
3. Autoscaling not enabled

The case of "ML APIs don't exist" is the easiest of all to account for, as it just needs a low level exception converting to something more friendly.